### PR TITLE
Enhancement: Boxes integration test

### PR DESCRIPTION
**Pull request by ahangsu**

### CODE REVIEW

File: .test-env [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # Configs for testing repo download:
 SDK_TESTING_URL="https://github.com/algorand/algorand-sdk-testing"
-SDK_TESTING_BRANCH="feature/box-storage"
+SDK_TESTING_BRANCH="ahangsu/indexer-integration-boxes"
```
Comment (ahangsu) [inline .test-env:3]: TODO: apply when `feature/box-storage` PR merged:
```suggestion
SDK_TESTING_BRANCH="feature/box-storage"
```

File: src/test/java/com/algorand/algosdk/integration/Clients.java [hunk 2]
```diff
@@ -9,10 +9,19 @@
 
 public class Clients {
     AlgodClient v2Client = null;
+
+    @Deprecated
     Map<Integer, IndexerClient> indexerClients = new HashMap<>();
```
Comment (ahangsu) [inline src/test/java/com/algorand/algosdk/integration/Clients.java:14]: shall I just remove this stuff? It should not get into SDK interface tho.

File: src/test/integration.tags [hunk 3]
```diff
@@ -3,6 +3,7 @@
 @applications
 @applications.boxes
 @applications.verified
+@applications.boxes_indexer_confirmed
```
Comment (tzaffi) [inline src/test/integration.tags:6]: same tag questions as in the other SDK's

File: src/test/java/com/algorand/algosdk/integration/Applications.java [hunk 4]
```diff
@@ -256,9 +258,15 @@ public void checkAccountData(
         assertThat(found).as("Couldn't find key '%s'", hasKey).isTrue();
     }
 
-    @Then("the contents of the box with name {string} in the current application should be {string}. If there is an error it is {string}.")
-    public void contentsOfBoxShouldBe(String encodedBoxName, String boxContents, String errStr) throws Exception {
-        Response<Box> boxResp = clients.v2Client.GetApplicationBoxByName(this.appId).name(encodedBoxName).execute();
+    @Then("according to {string}, the contents of the box with name {string} in the current application should be {string}. If there is an error it is {string}.")
+    public void contentsOfBoxShouldBe(String from_client, String encodedBoxName, String boxContents, String errStr) throws Exception {
```
Comment (tzaffi) [inline src/test/java/com/algorand/algosdk/integration/Applications.java:262]:
```suggestion
    public void contentsOfBoxShouldBe(String fromClient, String encodedBoxName, String boxContents, String errStr) throws Exception {
```
camel case for java

File: src/test/java/com/algorand/algosdk/integration/Clients.java (hunk 2 above)
Comment (tzaffi) [reply src/test/java/com/algorand/algosdk/integration/Clients.java:14]: Yeah, I recommend removing completely. We're only going to have one indexer. I missed this in a previous PR.

**Review by tzaffi — COMMENTED** [review]
tiny nits

File: src/test/integration.tags (hunk 3 above)
Comment (ahangsu) [reply src/test/integration.tags]: this one goes

File: src/test/java/com/algorand/algosdk/integration/Applications.java (hunk 4 above)
Comment (ahangsu) [reply src/test/java/com/algorand/algosdk/integration/Applications.java:262]: cAmelStyle

File: src/test/java/com/algorand/algosdk/integration/Applications.java [hunk 5]
```diff
@@ -295,17 +303,61 @@ private static boolean contains(byte[] elem, List<byte[]> xs) {
         return false;
     }
 
-    @Then("the current application should have the following boxes {string}.")
-    public void checkAppBoxes(String encodedBoxesRaw) throws Exception {
+    @Then("according to {string}, the current application should have the following boxes {string}.")
+    public void checkAppBoxes(String from_client, String encodedBoxesRaw) throws Exception {
```
[the remark this answers is not in the document]
Comment (ahangsu) [reply src/test/java/com/algorand/algosdk/integration/Applications.java]: good catch, my bad for not noticing.

**Review by tzaffi — APPROVED** [review]

File: src/test/java/com/algorand/algosdk/integration/Applications.java [hunk 6]
```diff
@@ -40,6 +40,8 @@ public class Applications {
     public String txId = null;
     public Long appId = 0L;
     public List<Long> rememberedAppIds = new ArrayList<>();
+    public TransactionParametersResponse sp = null;
```
Comment (algochoi) [inline src/test/java/com/algorand/algosdk/integration/Applications.java:43]: Are these parameters used?

File: src/test/java/com/algorand/algosdk/integration/Applications.java (hunk 4 above)
Comment (algochoi) [reply src/test/java/com/algorand/algosdk/integration/Applications.java:262]: It looks like this `from_client` is still in snake case 🐍  is this intentional?

File: src/test/java/com/algorand/algosdk/integration/Applications.java (hunk 6 above)
Comment (ahangsu) [reply src/test/java/com/algorand/algosdk/integration/Applications.java]: nope, good find.

File: src/test/java/com/algorand/algosdk/integration/Applications.java (hunk 4 above)
Comment (ahangsu) [reply src/test/java/com/algorand/algosdk/integration/Applications.java]: fixed.

**Review by algochoi — APPROVED** [review]
LGTM Pending tests!

**Pull request merged**